subsys/usb/usb_c/usbc_tc_src_states.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(usbc_stack, CONFIG_USBC_STACK_LOG_LEVEL);
+
+#include "usbc_stack.h"
+#include "usbc_tc_src_states_internal.h"
+
+/**
+ * @brief 4.5.2.2.7 Unattached.SRC State
+ *
+ * When in the Unattached.SRC state, the port is waiting to detect the
+ * presence of a Sink or an Accessory.
+ *
+ * Requirements:
+ *   1: The port shall not drive VBUS or VCONN.
+ *      NOTE: Implemented in the tc_attached_src_exit
+ *            function and initially set the tc_init function.
+ *
+ *   2: The port shall provide a separate Rp termination on the CC1 and
+ *      CC2 pins.
+ *      NOTE: Implemented in the tc_cc_rp super state.
+ */
+
+void tc_unattached_src_entry(void *obj)
+{
+	LOG_INF("Unattached.SRC");
+}
+
+void tc_unattached_src_run(void *obj)
+{
+	struct tc_sm_t *tc = (struct tc_sm_t *)obj;
+	const struct device *dev = tc->dev;
+
+	/*
+	 * Transition to AttachWait.SRC when:
+	 *   The SRC.Rd is detected on either CC1 or CC2 pin or
+	 *   SRC.Ra is detected on both CC1 and CC2 pins.
+	 *   NOTE: Audio Adapter Accessory Mode is not supported, so
+	 *   SRC.Ra will not be checked.
+	 */
+	if (tcpc_is_cc_at_least_one_rd(tc->cc1, tc->cc2)) {
+		tc_set_state(dev, TC_ATTACH_WAIT_SRC_STATE);
+	}
+}
+
+/**
+ * @brief 4.5.2.2.6 UnattachedWait.SRC State
+ *
+ * When in the UnattachedWait.SRC state, the port is discharging the CC pin
+ * that was providing VCONN in the previous Attached.SRC state.
+ *
+ * Requirements:
+ *  1: The port shall not enable VBUS or VCONN.
+ *     NOTE: Implemented in tc_attached_src_exit
+ *
+ *  2: The port shall continue to provide an Rp termination on the CC pin not
+ *     being discharged.
+ *     NOTE: Implemented in last TC source state.
```

Comment:
I thought this was controlled by setting Rp on entering the Rp super state.

Reply:
I've updated the comment